### PR TITLE
Untrusted competition

### PR DESCRIPTION
**Pull request by ratosh**

### CODE REVIEW

File: aiarena/core/models/competition.py [hunk 1]
```diff
@@ -62,6 +62,7 @@ class Competition(models.Model, LockableModelMixin):
     n_placements = models.IntegerField(default=0, validators=[MinValueValidator(0)], blank=True)
     # List of which bot races are playable in this competition. When left blank, all races are playable.
     playable_races = models.ManyToManyField(BotRace, blank=True)
+    trusted = models.BooleanField(default=True)
```
Comment (lladdy) [inline aiarena/core/models/competition.py:65]: Thoughts on renaming this to something else which indicates that it's not the competition that's "trusted", but that it requires trusted ACs?
Such as `require_trusted_infrastructure`, or `secure` - something like this?

File: aiarena/core/api/matches.py [hunk 2]
```diff
@@ -245,7 +245,7 @@ def _attempt_to_generate_new_round(competition: Competition):
                 .filter(competition=competition, active=True, division_num=participant1.division_num)
                 .exclude(id__in=already_processed_participants))
             for participant2 in active_participants_in_div:
-                Match.create(new_round, random.choice(active_maps), participant1.bot, participant2.bot, require_trusted_arenaclient=True)
+                Match.create(new_round, random.choice(active_maps), participant1.bot, participant2.bot, require_trusted_arenaclient=competition.trusted)
```
Comment (lladdy) [inline aiarena/core/api/matches.py:248]: We could choose this PR to be the one where we get rid of this `require_trusted_arenaclient` field and instead have it determined in `next_competition_match` call. Thoughts?

**Review by lladdy — COMMENTED** [review]
Looks good so far.
Not sure what's happening with github CI. I'll need to investigate when I get time.

File: aiarena/core/models/competition.py (hunk 1 above)
Comment (ratosh) [reply aiarena/core/models/competition.py:65]: Yes, It means that it requires trusted clients. We can rename it to `require_trusted_infrastructure`.

File: aiarena/core/api/matches.py (hunk 2 above)
Comment (ratosh) [reply aiarena/core/api/matches.py:248]: Having the trusted on match actually made this change really simple so I'd like to keep that. We can have:
- When we creating the match we store if that requires a trusted client or not (mostly for trusted competitions), changing the meaning to that particular match needs to run on trusted infra or not;
- When fetching matches we check if they're it requires trusted infra or are not publicly available;
We need to keep one eye on the number of non-public bots cos we could be reaching the limit of rounds when non-trusted clients are playing the matches for a small part of bots and starting new rounds to get more matches.

Comment (lladdy) [reply aiarena/core/api/matches.py:248]: > could be reaching the limit of rounds
I guess a community/untrusted ladder will be good in this respect.

> When we creating the match we store if that requires a trusted client or not
> When fetching matches we check if they're it requires trusted infra or are not publicly available;
I don't see the need to store it, unless we simply don't want to do the join to check the competition's trusted field when the match is dealt out?
I think from a code readability perspective, it'd be nice to have 1 method that does this check dynamically. The Match `require_trusted_arenaclient` field might be slightly confusing/misleading when there's also other things which will dictate whether the match requires a trusted AC.

Thoughts?

**Comment by lladdy** [comment]
@ratosh 
Can you please merge your migrations? Then the tests will run. I'll review after.
```
CommandError: Conflicting migrations detected; multiple leaf nodes in the migration graph: (0054_rename_plays_race_model_bot_plays_race, 0055_trusted_competition in core).
To fix them run 'python manage.py makemigrations --merge'
```

File: aiarena/core/models/match.py [hunk 3]
```diff
@@ -69,22 +69,6 @@ def create(round, map, bot1, bot2, requested_by=None,
...
                require_trusted_arenaclient=True):
         with transaction.atomic():
-            if bot1_use_data is None:
-                bot1_use_data = bot1.bot_data_enabled
-            if bot1_update_data is None:
-                bot1_update_data = bot1.bot_data_enabled
-            if bot2_use_data is None:
-                bot2_use_data = bot2.bot_data_enabled
-            if bot2_update_data is None:
-                bot2_update_data = bot2.bot_data_enabled
```
Comment (lladdy) [inline aiarena/core/models/match.py:79]: @ratosh it looks like removing these lines is causing the `use_bot_data` field for the MatchParticipations to be null.
We might be safe to add these back in without any other modification.

**Comment by lladdy** [comment]
@ratosh I modified an earlier migration that was causing a problem.

File: aiarena/core/models/match.py (hunk 3 above)
Comment (ratosh) [reply aiarena/core/models/match.py:81]: Reverted

**Pull request merged**